### PR TITLE
kubernetes: Shared images state on a project

### PR DESCRIPTION
**Pull request by stefwalter**
Show the shared images state on a project both on the dashboard, and in the projects page, and allow modifying that setting.

### CODE REVIEW

**Comment by stefwalter** [comment]
Needs tests.

**Comment by sub-mod** [comment]
duplication of tasks https://github.com/cockpit-project/cockpit/pull/4022 
https://github.com/cockpit-project/cockpit/pull/4023

**Comment by stefwalter** [comment]
I asked you about whether I should work on this, or whether it would duplicate efforts, and you said to go ahead.

Those pull requests have not been updated in over a week. Is there any chance of bringing those into a complete, reviewable, integration tested state shortly?

**Comment by petervo** [comment]
Makefile is missing some templates.

File: pkg/kubernetes/scripts/kube-client.js [hunk 1]
```diff
@@ -1156,6 +1158,30 @@
                 });
             }
 
+            function putResource(resource, body, config) {
```
Comment (petervo) [inline pkg/kubernetes/scripts/kube-client.js]: These are really similar, should they be generalized?

File: pkg/kubernetes/scripts/policy.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+                        name: name,
+                        namespace: namespace,
+                        subjects: subjects,
+                    },
+                    roleRef: {
+                        name: name
+                    }
+                };
+
+                console.log("creating", binding, namespace);
```
Comment (petervo) [inline pkg/kubernetes/scripts/policy.js]: Still needed?

File: pkg/kubernetes/scripts/registry.js [hunk 3]
```diff
@@ -29,6 +29,7 @@
         'kubernetes.app',
         'registry.images',
         'registry.projects',
+        'registry.policy',
```
Comment (petervo) [inline pkg/kubernetes/scripts/registry.js]: This is also needed in main.js

File: pkg/kubernetes/scripts/kube-client.js (hunk 1 above)
Comment (stefwalter) [inline pkg/kubernetes/scripts/kube-client.js]: Done.

File: pkg/kubernetes/scripts/registry.js (hunk 3 above)
Comment (stefwalter) [inline pkg/kubernetes/scripts/registry.js]: Done.

File: pkg/kubernetes/scripts/policy.js (hunk 2 above)
Comment (stefwalter) [inline pkg/kubernetes/scripts/policy.js]: Removed.

**Comment by stefwalter** [comment]
Tests done.

**Comment by sub-mod** [comment]
As discussed.Unfortunately some duplication of task has happened.Gonna live with it.
the plan is:
1)stefw to merge policy.js in this PR.
2)I will merge my changes into policy.js and add changes to common code(if there any from my side)
PRs https://github.com/cockpit-project/cockpit/pull/4022 https://github.com/cockpit-project/cockpit/pull/4023 to be rebased on top of https://github.com/cockpit-project/cockpit/pull/4078

**Pull request closed without merging**